forc-pkg/src/pkg.rs

```diff
@@ -590,7 +590,7 @@ impl GitReference {
     pub fn resolve(&self, repo: &git2::Repository) -> Result<git2::Oid> {
         // Find the commit associated with this tag.
         fn resolve_tag(repo: &git2::Repository, tag: &str) -> Result<git2::Oid> {
-            let refname = format!("refs/remotes/origin/tags/{}", tag);
```

Comment:
was this wrong before? is there documentation for this and should we make it a constant or something?

Reply:
The thing is we were fetching the package with an anonymous remote, which [cannot](https://docs.rs/git2/latest/git2/struct.Repository.html#method.remote_anonymous) be converted to a persisted remote. Since I needed the fetched package to be a valid git repo that we can interact with, later on, I changed that. After this change once `resolve` was failing and once I checked the fetched repo with `git show-ref`, I realized `refnames` changed. I think it is because of the change from `remote_anonymous` to `remote`.

I couldn't find why this is happening in the documentation of `git2`.